### PR TITLE
feat: retry URI creation

### PR DESCRIPTION
**Pull request by shaffeeullah**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -421,9 +427,39 @@ export class Upload extends Pumpify {
...
+              {
+                reason: e.response.statusText
+              }
+            ]
+          }
+          if (this.retryLimit > 0 && this.retryableErrorFn!(apiError)) {
+            throw new Error();
+          }
+          else {
+            return bail(new Error(`Failed with error ${e.response}`));
```
Comment (ddelgrosso1) [inline src/index.ts:451]: If possible, this should probably return an ApiError up to bail so that we can include the code as well as the message.

Comment (shaffeeullah) [reply src/index.ts]: i'm going to return `e` raw to preserve previous behavior

File: src/index.ts [hunk 2]
```diff
@@ -421,9 +427,41 @@ export class Upload extends Pumpify {
     if (this.origin) {
       reqOpts.headers!.Origin = this.origin;
     }
+    const uri = await retry(
```
Comment (shaffeeullah) [inline src/index.ts:430]: actual retry logic will be covered by conformance testing

File: src/index.ts [hunk 3]
```diff
@@ -211,8 +212,13 @@ export interface RetryOptions {
   retryableErrorFn?: (err: ApiError) => boolean;
 }
 
+export interface GoogleInnerError {
```
Comment (ddelgrosso1) [inline src/index.ts:215]: Any reason for the additional interface as opposed to just declaring `errors` (below) as `string[]`?

Comment (shaffeeullah) [reply src/index.ts:215]: yeah the compiler won't let me. It fails to map the logic below to `ApiError` without this

**Review by ddelgrosso1 — APPROVED** [review]

**Pull request merged**